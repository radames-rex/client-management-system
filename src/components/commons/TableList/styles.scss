.table {
	.tableCell {
		padding: 0.2188rem 0.625rem 0.2188rem 1rem;
		background-color: #6f7a89;
		color: #fff;
		font-weight: bold;
	}
}
