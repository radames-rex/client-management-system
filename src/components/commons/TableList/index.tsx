import React from "react";
import { Link } from "react-router-dom";
import {
	Paper,
	Table,
	TableBody,
	TableCell,
	TableHead,
	TableRow,
	IconButton
} from "@material-ui/core";
import EditIcon from "@material-ui/icons/Edit";
import DeleteIcon from "@material-ui/icons/Delete";

import { User } from "../../../store/ducks/users/types";
import "./styles.scss";

interface OwnProps {
	users: User[];
	deleteUsers: Function;
}

const renderCommands = (id: number, deleteUsers: Function) => {
	return (
		<>
			<Link to={`/edit/${id}`}>
				<IconButton aria-label="Edit">
					<EditIcon />
				</IconButton>
			</Link>

			<IconButton aria-label="Delete" onClick={() => deleteUsers(id)}>
				<DeleteIcon />
			</IconButton>
		</>
	);
};

export default function UserItem({ users, deleteUsers }: OwnProps) {
	return (
		<Paper>
			<Table className="table">
				<TableHead>
					<TableRow>
						<TableCell className="tableCell">Name</TableCell>
						<TableCell className="tableCell">Email</TableCell>
						<TableCell className="tableCell">CPF</TableCell>
						<TableCell className="tableCell">Phone Number</TableCell>
						<TableCell className="tableCell" />
					</TableRow>
				</TableHead>
				<TableBody>
					{users.map(user => (
						<TableRow key={user.id}>
							<TableCell>{user.name}</TableCell>
							<TableCell>{user.email}</TableCell>
							<TableCell>{user.cpf}</TableCell>
							<TableCell>{user.phone}</TableCell>
							<TableCell>{renderCommands(user.id, deleteUsers)}</TableCell>
						</TableRow>
					))}
				</TableBody>
			</Table>
		</Paper>
	);
}
