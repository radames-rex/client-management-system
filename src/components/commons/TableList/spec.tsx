import React from "react";
import { shallow } from "enzyme";
import TableList from "./index";

describe("<TableList /> component", () => {
	it("render correctly", () => {
		const wrapper = shallow(
			<TableList users={[]} deleteUsers={(__: any) => __} />
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper).toHaveLength(1);
	});
});
