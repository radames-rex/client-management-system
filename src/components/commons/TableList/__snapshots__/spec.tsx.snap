// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableList /> component render correctly 1`] = `ShallowWrapper {}`;
